probe-rs-cli-util/src/logging.rs

```diff
@@ -115,27 +192,179 @@ pub fn clear_progress_bar() {
+        release: Some(Cow::<'static>::Owned(release.to_string())),
+        #[cfg(debug_assertions)]
+        environment: Some(Cow::Borrowed("Development")),
+        #[cfg(not(debug_assertions))]
+        environment: Some(Cow::Borrowed("Production")),
+        default_integrations: false,
+        ..Default::default()
+    }
+}
+
+pub struct Metadata {
+    pub chip: Option<String>,
+    pub probe: Option<String>,
+    pub release: String,
+    pub commit: String,
+}
+
+/// Sets the metadata concerning the current probe-rs session on the sentry scope.
+fn set_metadata(metadata: &Metadata) {
+    sentry::configure_scope(|scope| {
+        metadata
+            .chip
+            .as_ref()
+            .map(|chip| scope.set_tag("chip", chip));
+        metadata
+            .probe
+            .as_ref()
+            .map(|probe| scope.set_tag("probe", probe));
+        scope.set_tag("commit", &metadata.commit);
+    })
+}
+
+fn print_uuid(uuid: Uuid) {
+    println(format!(
+        "  {} {} {}",
+        "Thank You!".cyan().bold(),
+        "Your error was reported successfully. If you don't mind, please open an issue on Github and include the UUID: ",
+        uuid
+    ));
+}
+
+/// Captures an std::error::Error with sentry and sends all previously captured logs.
+pub fn capture_error<E>(metadata: &Metadata, error: &E)
+where
+    E: Error + ?Sized,
+{
+    let _guard = sentry::init(sentry_config(metadata.release.clone()));
+    set_metadata(metadata);
+    send_logs();
+    let uuid = sentry::capture_error(error);
+    print_uuid(uuid);
+}
+
+/// Captures an anyhow error with sentry and sends all previously captured logs.
+pub fn capture_anyhow(metadata: &Metadata, error: &anyhow::Error) {
+    let _guard = sentry::init(sentry_config(metadata.release.clone()));
+    set_metadata(metadata);
+    send_logs();
+    let uuid = sentry::integrations::anyhow::capture_anyhow(error);
+    print_uuid(uuid);
+}
+
+/// Captures a panic with sentry and sends all previously captured logs.
+pub fn capture_panic(metadata: &Metadata, info: &PanicInfo<'_>) {
+    let _guard = sentry::init(sentry_config(metadata.release.clone()));
+    set_metadata(metadata);
+    send_logs();
+    let uuid = sentry::capture_event(PanicIntegration::new().event_from_panic_info(info));
+    print_uuid(uuid);
+}
+
+/// Ask for a line of text.
+fn text() -> std::io::Result<String> {
+    // Read up to the first newline or EOF.
+
+    let mut out = String::new();
+    std::io::stdin().read_line(&mut out)?;
+
+    // Only capture up to the first newline.
+    if let Some(mut newline) = out.find('\n') {
+        if newline > 0 && out.as_bytes()[newline - 1] == b'\r' {
+            newline -= 1;
+        }
+        out.truncate(newline);
+    }
+
+    Ok(out)
+}
+
+/// Displays the text to ask if the crash should be reported.
+pub fn ask_to_log_crash() -> bool {
+    if let Ok(var) = std::env::var("PROBE_RS_SENTRY") {
+        var == "true"
+    } else {
+        println(format!(
```

Comment:
I think placing the message in a constant would improve readability here.

Reply:
How would you store it? I mean, would you store the formatted version? Also, would you make it one string?

Reply:
Storing the message in a string, and maybe make a formatted print function. That formatted print function could also be used for the other formatted output.